### PR TITLE
Add stock movement

### PR DESCRIPTION
**Pull request by nimish-mehta**
1. Adds bought count and total count to variant.
2. Allow admin to piecemeal cancel line items
3. Update views to add to stock instead of directly changing the value.

### CODE REVIEW

File: apps/ex_shop/web/models/variant.ex [hunk 1]
```diff
@@ -44,4 +52,44 @@ defmodule ExShop.Variant do
...
+  defp update_total_quantity(model) do
+    quantity_to_add = model.changes[:add_count]
+    if quantity_to_add do
+      put_change(model, :total_quantity, model.model.total_quantity + quantity_to_add)
+    else
+      model
+    end
+  end
+
+  defp update_bought_quantity(model) do
```
Comment (pikender) [inline apps/ex_shop/web/models/variant.ex]: Can we break the method to be explicit about increment/decrement ??

File: apps/ex_shop/web/models/variant.ex [hunk 2]
```diff
@@ -44,4 +52,44 @@ defmodule ExShop.Variant do
     |> cast_attachments(params, ~w(), ~w(image))
     |> cast_assoc(:variant_option_values, required: true, with: &ExShop.VariantOptionValue.from_variant_changeset/2)
   end
+
+  def buy_changeset(model, params \\ :empty) do
+    model
+    |> cast(params, ~w(buy_count), ~w())
+    |> update_bought_quantity
```
Comment (pikender) [inline apps/ex_shop/web/models/variant.ex]: Can we rename this to state actual intent like increment_bought_count ?

File: apps/ex_shop/web/models/variant.ex [hunk 3]
```diff
@@ -44,4 +52,44 @@ defmodule ExShop.Variant do
...
+  end
+
+  def restocking_changeset(model, params) do
+    model
+    |> cast(params, ~w(restock_count), ~w())
+    |> update_bought_quantity
```
Comment (pikender) [inline apps/ex_shop/web/models/variant.ex]: :ditto: https://github.com/vinsol/ex-shop/pull/19/files#r54563190

File: apps/ex_shop/web/models/line_item.ex [hunk 4]
```diff
@@ -44,6 +49,33 @@ defmodule ExShop.LineItem do
     cast(model, Map.merge(params, %{total: cost}), ~w(total), ~w())
   end
 
+  def move_stock(%ExShop.LineItem{fullfilled: true} = line_item) do
+    remove_stock_from_variant(line_item)
```
Comment (pikender) [inline apps/ex_shop/web/models/line_item.ex]: Can we sync the 2 methods like remove_stock_from_variant and add_stock_to_variant

or 

take_stock_from_variant / move_stock_back_to_variant ??

Comment (nimish-mehta) [inline apps/ex_shop/web/models/line_item.ex]: `add_stock_to_variant` sounds ambiguous. and `take_stock_from_variant` and `remove_stock_from_variant` are almost synonyms. How about `acquire_stock_from_variant` and `restock_variant` ?

File: apps/ex_shop/web/models/variant.ex (hunk 1 above)
Comment (nimish-mehta) [inline apps/ex_shop/web/models/variant.ex]: Agreed. I have moved them to their separate methods.

File: apps/ex_shop/web/models/line_item.ex (hunk 4 above)
Comment (pikender) [inline apps/ex_shop/web/models/line_item.ex]: :+1:

**Pull request closed without merging**